sound/soc/sof/intel/hda.c

```diff
@@ -499,6 +499,28 @@ static const struct sof_intel_dsp_desc
 	return chip_info;
 }
 
+irqreturn_t hda_dsp_interrupt_handler(int irq, void *context)
+{
+	struct snd_sof_dev *sdev = context;
+
+	if (is_ipc_irq(sdev) || is_stream_irq(sdev))
+		return IRQ_WAKE_THREAD;
+
+	return IRQ_NONE;
+}
+
+irqreturn_t hda_dsp_interrupt_thread(int irq, void *context)
+{
+	struct snd_sof_dev *sdev = context;
+
+	if (is_ipc_irq(sdev))
```

Comment:
@bardliao is_ipc_irq() is called in both hardirq and softirq. This is otherwise ok, but it's a bit odd that is_ipc_irq() also has side-effects in that it clears HDA_DSP_ADSPIC_IPC. I'd prefer to call the function something else than "is_foo_irq()" as to me (maybe just ipc_irq()). To me, a function with is_something(), indicates the function does not have side-effects. The functions here do more. 

I guess this still works as the ADSPIS status bit remains high despite  ADSPIC set low. And this matches with the current logic we have.

Reply:
@kv2019i I had the same feeling but it's hard to move the mask outside of the spinlock, and taking the spinlock twice, once to read ADSPIC2 and once to clear ADSPIS2 is not that great either.

maybe adding a comment would help explain the hidden side effect.
or rename from is_ipc to check_ipc, which is let of a status return

Reply:
Thanks @kv2019i @plbossart  I will rename it and add a comment.

Reply:
@kv2019i @plbossart done. BTW, do you know how to disable stream irq? Or we can leave it as it is for now?